config/providers.go

```diff
@@ -12,11 +16,18 @@ type Providers struct {
 	// Providers that are allowed by policy or expetion must still be valid
 	// providers.
 	Except []string
-	// Trust is a list of peer IDs that are allowed with being required to be
-	// valid providers.
+	// Trust is a list of peer IDs that are allowed to register as providers
+	// without being required to be vefified on-chain
 	Trust []string
 	// PollInterval is the amount of time to wait without getting any updates
 	// from a provider, before sending a request for the latest advertisement.
 	// Values are a number ending in "s", "m", "h" for seconds. minutes, hours.
-	PollInterval string
+	PollInterval Duration
+
+	// DiscoveryTimeout is the maximum amount of time that the indexer will
+	// spend trying to verify a new provider.
+	DiscoveryTimeout Duration
```

Comment:
Discovery timeout is a configurable value, and when discovery does not complete within this time, the in-progress attempt is canceled and is considered a failure to discover the requested miner.  If the timeout is 0, then there is no discovery timeout and discovery will continue until failure or success, unless it hangs forever.

The config file is initialized with a value of 2 minutes unless otherwise specified.